runbotics/runbotics-orchestrator-ui/src/src-app/views/action/ActionDetails/ExternalActionForm.tsx

```diff
@@ -0,0 +1,91 @@
+import { ChangeEvent, FC } from 'react';
+
+import { TextField } from '@mui/material';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+
+import { IAction } from '#src-app/types/model/action.model';
+
+import { ExternalActionFormValidationState, isScriptNameValid, isValueEmpty } from '../action.utils';
+
+
+interface ExternalActionFormProps {
+    draft: IAction;
+    setDraft: (value: React.SetStateAction<IAction>) => void;
+    formValidationState: ExternalActionFormValidationState;
+    setFormValidationState: (value: React.SetStateAction<ExternalActionFormValidationState>) => void;
+}
+
+export const ExternalActionForm: FC<ExternalActionFormProps> = ({ draft, setDraft, formValidationState, setFormValidationState }) => {
+    const { translate } = useTranslations();
+
+    const handleChange = (field: string, e: ChangeEvent<HTMLInputElement>) => {
+        setDraft(prev => ({
+            ...prev,
+            [e.target.name]: e.target.value
+        }));
+
+        if (field === 'script') {
```

Comment:
I'd rather create two separate functions for each change event. Currently, we unnecessarily increase the complexity